src/main.cpp

```diff
@@ -11,19 +11,19 @@
 #include "render.hpp"
 
 // Function to save image data in ppm format
-template <int width, int height> void save_image(color* fb_data) {
+template <int width, int height> void save_image(std::array<color, (width * height)> & fb_data) {
```

Comment:
Here you could use a lambda.
While there is templated lambdas, it would be overkill, since you can pass the sizes as normal parameters
```suggestion
auto save_image = [] (int width, int height, auto fb) {
```

Reply:
If we pass width and height as parameter, is there an an interest to declare save_image as a lambda instead of a plain function ?